common/setups/rasr/util/decode.py

```diff
@@ -0,0 +1,52 @@
+__all__ = [
+    "BaseRecognitionParameters",
+    "PriorArgs",
+    "SearchJobArgs",
+    "Lattice2CtmArgs",
+    "ScorerArgs",
+    "OptimizeJobArgs",
+]
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, TypedDict
+
+from sisyphus import tk
+
+import i6_core.rasr as rasr
+
+from ..config.am_config import TransitionTdp, SilenceTdp
+
+
+class BaseRecognitionParameters(TypedDict):
+    am_scales: List[float]
+    lm_scales: List[float]
+    pronunciation_scales: List[float]
+    tdp_scales: List[float]
+    transition_tdp: List[TransitionTdp]
+    silence_tdp: List[SilenceTdp]
```

Comment:
```suggestion
    transition_tdps: List[TransitionTdp]
    silence_tdps: List[SilenceTdp]
```
plural to be more consistent.

Again, if you assume there could be different entries, the typing should be the base TDP class and not the derived class with hardcoded values.